xapian-letor/tests/api_letor.cc

```diff
@@ -0,0 +1,165 @@
+    termgenerator.index_text("Lions, Tigers and Bears", 1, "S");
+    termgenerator.index_text("This is the paragraph of interest. Tigers are "
+			     "massive beasts - I wouldn't want to meet a "
+			     "hungry one anywhere. Lions are scary even when "
+			     "lyin' down. Bears are scary even when bare. "
+			     "Together I suspect they'd be less scary, as the "
+			     "tigers, lions, and bears would all keep each "
+			     "other busy. On the other hand, bears don't live "
+			     "in the same continent as far as I know.", 1,
+			     "XD");
+    termgenerator.index_text("Lions, Tigers and Bears");
+    termgenerator.increase_termpos();
+    termgenerator.index_text("This is the paragraph of interest. Tigers are "
+			     "massive beasts - I wouldn't want to meet a "
+			     "hungry one anywhere. Lions are scary even when "
+			     "lyin' down. Bears are scary even when bare. "
+			     "Together I suspect they'd be less scary, as the "
+			     "tigers, lions, and bears would all keep each "
+			     "other busy. On the other hand, bears don't live "
+			     "in the same continent as far as I know.");
+    db.add_document(doc);
+}
+
+DEFINE_TESTCASE(listnet_ranker, generated) {
+    Xapian::ListNETRanker ranker;
+    string a = get_database_path("apitest_ranker1",
+				       populate_db);
+    Xapian::Enquire enquire((Xapian::Database(a)));
+    enquire.set_query(Xapian::Query("lions"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    Xapian::prepare_training_file(a, "testdata/query.txt",
+				  "testdata/qrel.txt", 10,
+				  "training-data.txt");
+    ranker.set_database_path(a);
+    ranker.set_query(Xapian::Query("lions"));
+    ranker.train_model("training-data.txt", "");
+    Xapian::docid doc1 = *mymset[0];
+    Xapian::docid doc2 = *mymset[1];
+    ranker.rank(mymset, "");
+    TEST_EQUAL(doc2, *mymset[0]);
+    TEST_EQUAL(doc1, *mymset[1]);
+
+    return true;
+}
+
+DEFINE_TESTCASE(svm_ranker, generated) {
+    Xapian::SVMRanker ranker;
+    string a = get_database_path("apitest_ranker1",
+				       populate_db);
+    Xapian::Enquire enquire((Xapian::Database(a)));
+    enquire.set_query(Xapian::Query("lions"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    Xapian::prepare_training_file(a, "testdata/query.txt",
+				  "testdata/qrel.txt", 10,
+				  "training-data.txt");
+    ranker.set_database_path(a);
+    ranker.set_query(Xapian::Query("lions"));
+    ranker.train_model("training-data.txt", "");
+    Xapian::docid doc1 = *mymset[0];
+    Xapian::docid doc2 = *mymset[1];
+    ranker.rank(mymset, "");
+    TEST_EQUAL(doc2, *mymset[0]);
+    TEST_EQUAL(doc1, *mymset[1]);
+
+    return true;
+}
+
+DEFINE_TESTCASE(createfeaturevector, generated)
+{
+    Xapian::FeatureList fl;
+    Xapian::Database db = get_database("apitest_ranker1",
+				       populate_db);
+    Xapian::Enquire enquire(db);
+    enquire.set_query(Xapian::Query("lions"));
+    Xapian::MSet mset;
+    TEST(mset.empty());
```

Comment:
I mean specifically the code just above, where you create an empty `MSet`, then test that it's empty:

```
Xapian::MSet mset;
TEST(mset.empty());
```

That seems a test which belongs in the xapian-core testsuite rather than here, which leads me to wonder if you meant to write something different?